### ISSUE TITLE
Crash on Delete

### ISSUE THREAD

**Issue by ibigpapa** [heketi/heketi#1270]
### Kind of issue

> Uncomment only one of these:
> 
Bug

### Observed behavior
Attempted to delete a volume using `http://A.B.C.D:P/volumes/{id}` and heketi rest api server crashed. 
Fails to start back up as well.
Crash logs (Before Restart)
```
[negroni] Started DELETE /volumes/d8037f7f552fa3890cd6f408f755ead0
unexpected fault address 0x7f4c94f9e000
fatal error: fault
[signal SIGBUS: bus error code=0x2 addr=0x7f4c94f9e000 pc=0xdca927]

goroutine 244903 [running]:
runtime.throw(0x189c119, 0x5)
        /usr/lib/golang/src/runtime/panic.go:605 +0x95 fp=0xc4205ef340 sp=0xc4205ef320 pc=0x42cba5
runtime.sigpanic()
        /usr/lib/golang/src/runtime/signal_unix.go:364 +0x29d fp=0xc4205ef390 sp=0xc4205ef340 pc=0x442aed
github.com/heketi/heketi/vendor/github.com/boltdb/bolt.(*freelist).free(0xc4201f05a0, 0x25, 0x7f4c94f9e000)
        /build/golang/src/github.com/heketi/heketi/vendor/github.com/boltdb/bolt/freelist.go:108 +0x37 fp=0xc4205ef450 sp=0xc4205ef390 pc=0xdca927
github.com/heketi/heketi/vendor/github.com/boltdb/bolt.(*Tx).Commit(0xc42074e2a0, 0x0, 0x0)
        /build/golang/src/github.com/heketi/heketi/vendor/github.com/boltdb/bolt/tx.go:176 +0x1fe fp=0xc4205ef5a8 sp=0xc4205ef450 pc=0xdd072e
github.com/heketi/heketi/vendor/github.com/boltdb/bolt.(*DB).Update(0xc4203a2b40, 0xc4203729a0, 0x0, 0x0)
        /build/golang/src/github.com/heketi/heketi/vendor/github.com/boltdb/bolt/db.go:602 +0xf2 fp=0xc4205ef5f8 sp=0xc4205ef5a8 pc=0xdc8842
github.com/heketi/heketi/apps/glusterfs.(*VolumeDeleteOperation).Build(0xc420133260, 0x18a4262, 0xd)
        /build/golang/src/github.com/heketi/heketi/apps/glusterfs/operations.go:389 +0x6f fp=0xc4205ef630 sp=0xc4205ef5f8 pc=0x14c914f
github.com/heketi/heketi/apps/glusterfs.AsyncHttpOperation(0xc420081680, 0x7f4c94fb71b0, 0xc42065a900, 0xc420140f00, 0x2550c20, 0xc420133260, 0xc4203f9a40, 0xc42037b600)
        /build/golang/src/github.com/heketi/heketi/apps/glusterfs/operations.go:1169 +0x77 fp=0xc4205ef6e8 sp=0xc4205ef630 pc=0x14cc027
github.com/heketi/heketi/apps/glusterfs.(*App).VolumeDelete(0xc420081680, 0x7f4c94fb71b0, 0xc42065a900, 0xc420140f00)
        /build/golang/src/github.com/heketi/heketi/apps/glusterfs/app_volume.go:277 +0x17c fp=0xc4205ef780 sp=0xc4205ef6e8 pc=0x14aca9c
github.com/heketi/heketi/apps/glusterfs.(*App).VolumeDelete-fm(0x7f4c94fb71b0, 0xc42065a900, 0xc420140f00)
        /build/golang/src/github.com/heketi/heketi/apps/glusterfs/app.go:459 +0x48 fp=0xc4205ef7b0 sp=0xc4205ef780 pc=0x14f7a18
net/http.HandlerFunc.ServeHTTP(0xc4202279f0, 0x7f4c94fb71b0, 0xc42065a900, 0xc420140f00)
        /usr/lib/golang/src/net/http/server.go:1918 +0x44 fp=0xc4205ef7d8 sp=0xc4205ef7b0 pc=0x6766f4
github.com/heketi/heketi/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc420081900, 0x7f4c94fb71b0, 0xc42065a900, 0xc420140f00)
        /build/golang/src/github.com/heketi/heketi/vendor/github.com/gorilla/mux/mux.go:114 +0xdc fp=0xc4205ef8d8 sp=0xc4205ef7d8 pc=0x6c409c
github.com/heketi/heketi/vendor/github.com/urfave/negroni.Wrap.func1(0x7f4c94fb71b0, 0xc42065a900, 0xc420140d00, 0xc42065ce40)
        /build/golang/src/github.com/heketi/heketi/vendor/github.com/urfave/negroni/negroni.go:41 +0x4d fp=0xc4205ef908 sp=0xc4205ef8d8 pc=0x71d46d
github.com/heketi/heketi/vendor/github.com/urfave/negroni.HandlerFunc.ServeHTTP(0xc42043f060, 0x7f4c94fb71b0, 0xc42065a900, 0xc420140d00, 0xc42065ce40)
        /build/golang/src/github.com/heketi/heketi/vendor/github.com/urfave/negroni/negroni.go:24 +0x4e fp=0xc4205ef938 sp=0xc4205ef908 pc=0x71c24e
github.com/heketi/heketi/vendor/github.com/urfave/negroni.middleware.ServeHTTP(0x253c4e0, 0xc42043f060, 0xc42043f0c0, 0x7f4c94fb71b0, 0xc42065a900, 0xc420140d00)
        /build/golang/src/github.com/heketi/heketi/vendor/github.com/urfave/negroni/negroni.go:33 +0xa5 fp=0xc4205ef988 sp=0xc4205ef938 pc=0x71c305
github.com/heketi/heketi/vendor/github.com/urfave/negroni.(middleware).ServeHTTP-fm(0x7f4c94fb71b0, 0xc42065a900, 0xc420140d00)
        /build/golang/src/github.com/heketi/heketi/vendor/github.com/urfave/negroni/negroni.go:33 +0x60 fp=0xc4205ef9c8 sp=0xc4205ef988 pc=0x71db50
github.com/heketi/heketi/vendor/github.com/urfave/negroni.(*Logger).ServeHTTP(0xc420158380, 0x7f4c94fb71b0, 0xc42065a900, 0xc420140d00, 0xc42065ce20)
        /build/golang/src/github.com/heketi/heketi/vendor/github.com/urfave/negroni/logger.go:25 +0x16d fp=0xc4205efaa0 sp=0xc4205ef9c8 pc=0x71c00d
github.com/heketi/heketi/vendor/github.com/urfave/negroni.middleware.ServeHTTP(0x2534820, 0xc420158380, 0xc42043f0a0, 0x7f4c94fb71b0, 0xc42065a900, 0xc420140d00)
        /build/golang/src/github.com/heketi/heketi/vendor/github.com/urfave/negroni/negroni.go:33 +0xa5 fp=0xc4205efaf0 sp=0xc4205efaa0 pc=0x71c305
github.com/heketi/heketi/vendor/github.com/urfave/negroni.(middleware).ServeHTTP-fm(0x7f4c94fb71b0, 0xc42065a900, 0xc420140d00)
        /build/golang/src/github.com/heketi/heketi/vendor/github.com/urfave/negroni/negroni.go:33 +0x60 fp=0xc4205efb30 sp=0xc4205efaf0 pc=0x71db50
github.com/heketi/heketi/vendor/github.com/urfave/negroni.(*Recovery).ServeHTTP(0xc420120220, 0x7f4c94fb71b0, 0xc42065a900, 0xc420140d00, 0xc42065ce00)
        /build/golang/src/github.com/heketi/heketi/vendor/github.com/urfave/negroni/recovery.go:45 +0x82 fp=0xc4205efb68 sp=0xc4205efb30 pc=0x71cec2
github.com/heketi/heketi/vendor/github.com/urfave/negroni.middleware.ServeHTTP(0x25348a0, 0xc420120220, 0xc42043f080, 0x7f4c94fb71b0, 0xc42065a900, 0xc420140d00)
        /build/golang/src/github.com/heketi/heketi/vendor/github.com/urfave/negroni/negroni.go:33 +0xa5 fp=0xc4205efbb8 sp=0xc4205efb68 pc=0x71c305
github.com/heketi/heketi/vendor/github.com/urfave/negroni.(*Negroni).ServeHTTP(0xc4201f0420, 0x254c560, 0xc42074e000, 0xc420140d00)
        /build/golang/src/github.com/heketi/heketi/vendor/github.com/urfave/negroni/negroni.go:73 +0x100 fp=0xc4205efc18 sp=0xc4205efbb8 pc=0x71c650
github.com/heketi/heketi/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc4200817c0, 0x254c560, 0xc42074e000, 0xc420140d00)
        /build/golang/src/github.com/heketi/heketi/vendor/github.com/gorilla/mux/mux.go:114 +0xdc fp=0xc4205efd18 sp=0xc4205efc18 pc=0x6c409c
net/http.serverHandler.ServeHTTP(0xc42024f860, 0x254c560, 0xc42074e000, 0xc420140900)
        /usr/lib/golang/src/net/http/server.go:2619 +0xb4 fp=0xc4205efd48 sp=0xc4205efd18 pc=0x679314
net/http.(*conn).serve(0xc4203fa8c0, 0x254d7e0, 0xc42065a7c0)
        /usr/lib/golang/src/net/http/server.go:1801 +0x71d fp=0xc4205effc8 sp=0xc4205efd48 pc=0x6755cd
runtime.goexit()
        /usr/lib/golang/src/runtime/asm_amd64.s:2337 +0x1 fp=0xc4205effd0 sp=0xc4205effc8 pc=0x45b131
created by net/http.(*Server).Serve
        /usr/lib/golang/src/net/http/server.go:2720 +0x288

goroutine 1 [select, 4290 minutes]:
main.main()
        /build/golang/src/github.com/heketi/heketi/main.go:387 +0xa13

goroutine 19 [syscall, 4290 minutes]:
os/signal.signal_recv(0x0)
        /usr/lib/golang/src/runtime/sigqueue.go:131 +0xa6
os/signal.loop()
        /usr/lib/golang/src/os/signal/signal_unix.go:22 +0x22
created by os/signal.init.0
        /usr/lib/golang/src/os/signal/signal_unix.go:28 +0x41

goroutine 20 [chan receive, 2 minutes]:
github.com/heketi/heketi/vendor/github.com/golang/glog.(*loggingT).flushDaemon(0x25b4ea0)
        /build/golang/src/github.com/heketi/heketi/vendor/github.com/golang/glog/glog.go:879 +0x9f
created by github.com/heketi/heketi/vendor/github.com/golang/glog.init.0
        /build/golang/src/github.com/heketi/heketi/vendor/github.com/golang/glog/glog.go:410 +0x203

goroutine 52 [select]:
github.com/heketi/heketi/apps/glusterfs.(*NodeHealthCache).Monitor.func1(0xc420056600, 0xc42021e8a0, 0xc4200565c0, 0xc420114180)
        /build/golang/src/github.com/heketi/heketi/apps/glusterfs/health_cache.go:118 +0x1aa
created by github.com/heketi/heketi/apps/glusterfs.(*NodeHealthCache).Monitor
        /build/golang/src/github.com/heketi/heketi/apps/glusterfs/health_cache.go:114 +0xc3

goroutine 54 [select, 4290 minutes, locked to thread]:
runtime.gopark(0x191c6c0, 0x0, 0x189d865, 0x6, 0x18, 0x1)
        /usr/lib/golang/src/runtime/proc.go:287 +0x12c
runtime.selectgo(0xc420039f50, 0xc42021f3e0)
        /usr/lib/golang/src/runtime/select.go:395 +0x1149
runtime.ensureSigM.func1()
        /usr/lib/golang/src/runtime/signal_unix.go:511 +0x220
runtime.goexit()
        /usr/lib/golang/src/runtime/asm_amd64.s:2337 +0x1

goroutine 55 [IO wait]:
internal/poll.runtime_pollWait(0x7f4c95087eb0, 0x72, 0xffffffffffffffff)
        /usr/lib/golang/src/runtime/netpoll.go:173 +0x57
internal/poll.(*pollDesc).wait(0xc4202c3698, 0x72, 0xc42049fc00, 0x0, 0x0)
        /usr/lib/golang/src/internal/poll/fd_poll_runtime.go:85 +0xae
internal/poll.(*pollDesc).waitRead(0xc4202c3698, 0xffffffffffffff00, 0x0, 0x0)
        /usr/lib/golang/src/internal/poll/fd_poll_runtime.go:90 +0x3d
internal/poll.(*FD).Accept(0xc4202c3680, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        /usr/lib/golang/src/internal/poll/fd_unix.go:335 +0x1e2
net.(*netFD).accept(0xc4202c3680, 0x191c770, 0xc42049fd88, 0x40320b)
        /usr/lib/golang/src/net/fd_unix.go:238 +0x42
net.(*TCPListener).accept(0xc4201592f8, 0x1692200, 0xc42049fdb8, 0x401fa7)
        /usr/lib/golang/src/net/tcpsock_posix.go:136 +0x2e
net.(*TCPListener).AcceptTCP(0xc4201592f8, 0xc42049fe00, 0xc42049fe08, 0xc42049fdf8)
        /usr/lib/golang/src/net/tcpsock.go:234 +0x49
net/http.tcpKeepAliveListener.Accept(0xc4201592f8, 0x191c0a0, 0xc4203fa8c0, 0x254d8a0, 0xc4203f8ae0)
        /usr/lib/golang/src/net/http/server.go:3120 +0x2f
net/http.(*Server).Serve(0xc42024f860, 0x254d1e0, 0xc4201592f8, 0x0, 0x0)
        /usr/lib/golang/src/net/http/server.go:2695 +0x1b2
net/http.(*Server).ListenAndServe(0xc42024f860, 0xc42024f860, 0xc42024b6e0)
        /usr/lib/golang/src/net/http/server.go:2636 +0xa9
net/http.ListenAndServe(0xc42024b6e0, 0x5, 0x2534760, 0xc4200817c0, 0x4, 0xc42024b6e0)
        /usr/lib/golang/src/net/http/server.go:2882 +0x7f
main.main.func2(0xc420229d50, 0xc420226660, 0xc4200817c0, 0xc42021f4a0)
        /build/golang/src/github.com/heketi/heketi/main.go:378 +0x2fd
created by main.main
        /build/golang/src/github.com/heketi/heketi/main.go:371 +0x965

goroutine 244904 [IO wait]:
internal/poll.runtime_pollWait(0x7f4c95087c70, 0x72, 0x0)
        /usr/lib/golang/src/runtime/netpoll.go:173 +0x57
internal/poll.(*pollDesc).wait(0xc42037b098, 0x72, 0xffffffffffffff00, 0x253d920, 0x2531e20)
        /usr/lib/golang/src/internal/poll/fd_poll_runtime.go:85 +0xae
internal/poll.(*pollDesc).waitRead(0xc42037b098, 0xc4203f9700, 0x1, 0x1)
        /usr/lib/golang/src/internal/poll/fd_poll_runtime.go:90 +0x3d
internal/poll.(*FD).Read(0xc42037b080, 0xc4203f9751, 0x1, 0x1, 0x0, 0x0, 0x0)
        /usr/lib/golang/src/internal/poll/fd_unix.go:126 +0x18a
net.(*netFD).Read(0xc42037b080, 0xc4203f9751, 0x1, 0x1, 0xc42072e5a0, 0x0, 0x253aa01)
        /usr/lib/golang/src/net/fd_unix.go:202 +0x52
net.(*conn).Read(0xc42066c078, 0xc4203f9751, 0x1, 0x1, 0x0, 0x0, 0x0)
        /usr/lib/golang/src/net/net.go:176 +0x6d
net/http.(*connReader).backgroundRead(0xc4203f9740)
        /usr/lib/golang/src/net/http/server.go:660 +0x62
created by net/http.(*connReader).startBackgroundRead
        /usr/lib/golang/src/net/http/server.go:656 +0xd8
```

K8s Restart events
```
Warning  
FailedMount  
29m (x22 over 1h)  
kubelet, A.B.C.D
Unable to mount volumes for pod "heketi-677d887c68-m2wdl_default(69b7b2d8-86ee-11e8-851d-005056851d19)": timeout expired waiting for volumes to attach or mount for pod "default"/"heketi-677d887c68-m2wdl". list of unmounted volumes=[db]. list of unattached volumes=[db config heketi-service-account-token-asdf]

Warning
FailedMount  3m (x43 over 1h)
kubelet, A.B.C.D
MountVolume.SetUp failed for volume "db" : stat /var/lib/kubelet/pods/69b7b2d8-86ee-11e8-851d-005056851d19/volumes/kubernetes.io~glusterfs/db: transport endpoint is not connected
```

### Expected/desired behavior
Delete the volume i asked it to and not crash.

### Details on how to reproduce (minimal and precise)

- create a storage class using `kubectl create -f sc.yml`
- create a pvc using `kubectl create -f pvc.yml`
- delete storage class using `kubectl delete -f sc.yml`
- access volumes via api `curl http://A.B.C.D:P/volumes`  
- identify volume via api `curl http://A.B.C.D:P/volumes/{id}`
- delete volume via api `curl -X DELETE http://A.B.C.D:P/volumes/{id}`

sc.yml
```yaml
apiVersion: storage.k8s.io/v1beta1
kind: StorageClass
metadata:
  name: glusterfs-storage
provisioner: kubernetes.io/glusterfs
parameters:
  resturl: "http://A.B.C.D:P"
  volumetype: "none"
```

pvc.yml
```yaml
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: pvc-glusterfs-storage
  annotations:
    volume.beta.kubernetes.io/storage-class: glusterfs-storage
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 5Gi
```

### Information about the environment:

- Heketi version used (e.g. v6.0.0 or master): master
- Operating system used: Ubuntu 16.04
- Heketi compiled from sources, as a package (rpm/deb), or container: container
- If container, which container image: heketi/heketi:dev (from gk-deploy master branch)
- Using kubernetes, openshift, or direct install: Kubernetes
- If kubernetes/openshift, is gluster running inside kubernetes/openshift or outside: Outside
- If kubernetes/openshift, how was it deployed (gk-deploy, openshift-ansible, other, custom):gk-deploy

**Comment by phlogistonjohn** [comment]
Hi @ibigpapa if you could would you please try retesting your case with the recently released Heketi v8.0.0? While I do not think we aimed to handle this exact error condition the internals around volume delete have changed quite a bit and I'd like to determine if your issue is still reproducible.
Thanks.

**Comment by phlogistonjohn** [comment]
Stale issue. Closing.
